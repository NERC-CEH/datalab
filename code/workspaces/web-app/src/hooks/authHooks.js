import useShallowSelector from './useShallowSelector';
import authSelectors from '../selectors/authSelectors';

const { currentUserId, currentUserPermissions, currentUserTokens } = authSelectors;

export const useCurrentUserId = () => useShallowSelector(currentUserId);
export const useCurrentUserPermissions = () => useShallowSelector(currentUserPermissions);
export const useCurrentUserTokens = () => useShallowSelector(currentUserTokens);
