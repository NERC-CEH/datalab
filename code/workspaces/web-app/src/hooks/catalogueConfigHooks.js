import useShallowSelector from './useShallowSelector';
import catalogueConfigSelectors from '../selectors/catalogueConfigSelectors';

export const useCatalogueConfig = () => useShallowSelector(catalogueConfigSelectors.catalogueConfig);
export const useCatalogueAvailable = () => useShallowSelector(catalogueConfigSelectors.catalogueAvailable);
