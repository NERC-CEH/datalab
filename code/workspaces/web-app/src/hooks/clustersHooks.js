import useShallowSelector from './useShallowSelector';
import clustersSelectors from '../selectors/clustersSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useClusters = () => useShallowSelector(clustersSelectors.clusters);
export const useClustersByType = (type) => {
  const clusters = useClusters();

  if (!type) {
    return clusters;
  }

  return {
    ...clusters,
    value: clusters.value.filter(cluster => cluster.type === type),
  };
};
