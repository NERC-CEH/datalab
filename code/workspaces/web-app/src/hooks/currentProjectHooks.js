import useShallowSelector from './useShallowSelector';
import currentProjectSelectors from '../selectors/currentProjectSelectors';

const { currentProject, currentProjectKey } = currentProjectSelectors;

export const useCurrentProject = () => useShallowSelector(currentProject);
export const useCurrentProjectKey = () => useShallowSelector(currentProjectKey);
