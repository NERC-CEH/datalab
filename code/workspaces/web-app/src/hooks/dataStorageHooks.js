import useShallowSelector from './useShallowSelector';
import dataStorageSelectors from '../selectors/dataStorageSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useDataStorageArray = () => useShallowSelector(dataStorageSelectors.dataStorageArray);
export const useDataStorageForUserInProject = (userId, projectKey) => {
  const dataStores = useDataStorageArray();
  return {
    ...dataStores,
    value: dataStores.value.filter(store => store.projectKey === projectKey && store.users.includes(userId)),
  };
};
