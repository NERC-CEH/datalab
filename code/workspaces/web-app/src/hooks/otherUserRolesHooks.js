import useShallowSelector from './useShallowSelector';
import otherUserRolesSelectors from '../selectors/otherUserRolesSelectors';

// eslint-disable-next-line import/prefer-default-export
export const useOtherUserRoles = () => useShallowSelector(otherUserRolesSelectors.otherUserRoles);
