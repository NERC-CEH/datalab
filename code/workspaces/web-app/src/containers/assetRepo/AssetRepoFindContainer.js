import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import sortByName from '../../components/common/sortByName';
import PromisedContentWrapper from '../../components/common/PromisedContentWrapper';
import Pagination from '../../components/stacks/Pagination';
import AssetMultiSelect from '../../components/common/form/AssetMultiSelect';
import assetRepoActions from '../../actions/assetRepoActions';
import { useAssetRepo } from '../../hooks/assetRepoHooks';
import AssetAccordion from '../../components/assetRepo/AssetAccordion';

const useStyles = makeStyles(theme => ({
  showControls: {
    display: 'flex',
    alignItems: 'center',
  },
  showText: {
    fontWeight: '400',
    marginRight: theme.spacing(4),
  },
  placeholderCard: {
    width: '100%',
    height: 70,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    borderTop: `1px solid ${theme.palette.divider}`,
    borderBottom: `1px solid ${theme.palette.divider}`,
  },
}));

function AssetRepoFindContainer({ userPermissions }) {
  const dispatch = useDispatch();
  const [selectedAssets, setSelectedAssets] = useState([]);
  const classes = useStyles();
  const assetRepo = useAssetRepo();
  const shownAssets = (selectedAssets && selectedAssets.length > 0) ? selectedAssets : sortByName(assetRepo.value.assets);

  useEffect(() => {
    dispatch(assetRepoActions.loadAllAssets());
  }, [dispatch]);

  const renderedAssets = shownAssets && shownAssets.length > 0
    ? shownAssets.map(asset => <AssetAccordion key={asset.assetId} asset={asset}/>)
    : [<div className={classes.placeholderCard} key={'placeholder-card'}>
      <Typography variant="body1">No assets to display.</Typography>
    </div>];

  const assetsInput = {
    onChange: (val) => { setSelectedAssets(val); },
    value: selectedAssets,
  };

  return (
    <>
      <AssetMultiSelect input={assetsInput} showAllAssets />
      <PromisedContentWrapper fetchingClassName={classes.placeholderCard} promise={assetRepo}>
        <Pagination items={renderedAssets} itemsPerPage={5} />
      </PromisedContentWrapper>
    </>
  );
}

export default AssetRepoFindContainer;
