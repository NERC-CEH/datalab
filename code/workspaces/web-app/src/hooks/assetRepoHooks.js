import useShallowSelector from './useShallowSelector';
import assetRepoSelectors from '../selectors/assetRepoSelectors';
import sortByName from '../components/common/sortByName';
import { PUBLIC } from '../components/assetRepo/assetVisibilities';

const useAssetRepoUnsorted = () => useShallowSelector(assetRepoSelectors.assetRepo);

export const useAssetRepo = () => {
  const assetRepo = useAssetRepoUnsorted();
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(assetRepo.value.assets),
    },
  };
};

const assetVisible = (projectKey, asset) => asset.fileLocation
  && (asset.visible === PUBLIC
  || (asset.projects && asset.projects.map(project => project.key).includes(projectKey)));

export const useVisibleAssets = (projectKey) => {
  const assetRepo = useAssetRepoUnsorted();
  const visibleAssets = assetRepo.value.assets
    ? assetRepo.value.assets.filter(asset => assetVisible(projectKey, asset))
    : [];
  return {
    ...assetRepo,
    value:
    {
      ...assetRepo.value,
      assets: sortByName(visibleAssets),
    },
  };
};
