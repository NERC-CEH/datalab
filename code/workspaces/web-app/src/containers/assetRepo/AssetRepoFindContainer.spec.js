import React from 'react';
import * as ReactRedux from 'react-redux';
import { shallow } from 'enzyme';
import AssetRepoFindContainer from './AssetRepoFindContainer';
import { useAssetRepo } from '../../hooks/assetRepoHooks';

jest.mock('react-router');
jest.mock('../../hooks/assetRepoHooks');

const asset = {
  name: 'asset name',
  version: 'asset version',
  assetId: 'asset-1',
  fileLocation: 'file location',
  masterUrl: 'master URL',
  owners: [{ userId: 'owner-1', name: 'name 1' }, { userId: 'owner-2', name: 'name 2' }],
  visibility: 'BY_PROJECT',
  projects: [{ projectKey: 'project-1', name: 'project 1' }, { projectKey: 'project-2', name: 'project 2' }],
};
const assetRep = { fetching: false, error: null, value: { assets: [asset] } };

describe('AssetRepoFindContainer', () => {
  const shallowRender = () => shallow(<AssetRepoFindContainer />);

  beforeEach(() => {
    const dispatchMock = jest.fn().mockName('dispatch');
    jest.spyOn(ReactRedux, 'useDispatch').mockImplementation(() => dispatchMock);

    useAssetRepo.mockReturnValue(assetRep);
  });

  it('renders to match snapshot passing correct props to children', () => {
    expect(shallowRender()).toMatchSnapshot();
  });
});
