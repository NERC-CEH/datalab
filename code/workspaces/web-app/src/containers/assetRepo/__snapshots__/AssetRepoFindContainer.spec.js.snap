// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetRepoFindContainer renders to match snapshot passing correct props to children 1`] = `
<Fragment>
  <AssetMultiSelect
    input={
      Object {
        "onChange": [Function],
        "value": Array [],
      }
    }
    showAllAssets={true}
  />
  <PromisedContentWrapper
    fetchingClassName="makeStyles-placeholderCard-3"
    promise={
      Object {
        "error": null,
        "fetching": false,
        "value": Object {
          "assets": Array [
            Object {
              "assetId": "asset-1",
              "fileLocation": "file location",
              "masterUrl": "master URL",
              "name": "asset name",
              "owners": Array [
                Object {
                  "name": "name 1",
                  "userId": "owner-1",
                },
                Object {
                  "name": "name 2",
                  "userId": "owner-2",
                },
              ],
              "projects": Array [
                Object {
                  "name": "project 1",
                  "projectKey": "project-1",
                },
                Object {
                  "name": "project 2",
                  "projectKey": "project-2",
                },
              ],
              "version": "asset version",
              "visibility": "BY_PROJECT",
            },
          ],
        },
      }
    }
  >
    <Pagination
      items={
        Array [
          <AssetAccordion
            asset={
              Object {
                "assetId": "asset-1",
                "fileLocation": "file location",
                "masterUrl": "master URL",
                "name": "asset name",
                "owners": Array [
                  Object {
                    "name": "name 1",
                    "userId": "owner-1",
                  },
                  Object {
                    "name": "name 2",
                    "userId": "owner-2",
                  },
                ],
                "projects": Array [
                  Object {
                    "name": "project 1",
                    "projectKey": "project-1",
                  },
                  Object {
                    "name": "project 2",
                    "projectKey": "project-2",
                  },
                ],
                "version": "asset version",
                "visibility": "BY_PROJECT",
              }
            }
          />,
        ]
      }
      itemsPerPage={5}
    />
  </PromisedContentWrapper>
</Fragment>
`;
